### DISCUSSION TITLE
Controlled 컴포넌트가 무엇인가요?

### DISCUSSION THREAD

**Discussion by javascriptbiggosoo** [wanted-9th-3team/frontend-study#76] [프론트 기술 질문 공간]
Controlled vs Uncontrolled Components

**Comment by michoball** [comment]
Controlled component : 사용자의 입력을 기반으로  컴포넌트의 state를 관리합니다.  

ex) useState를 이용해 input onChange를 다룸

Uncontrolled component: submit 과 같은 행동을 해야만 값을 다룰 수 있습니다. 

ex) useRef 를 이용해 onSubmit시 상 값을 다룸

**Comment by WooGie911** [comment]
제어 컴포넌트(Controlled Component)는 사용자의 입력을 기반으로 자신의 state를 관리하고 업데이트합니다.
즉 입력 값이 생길 때마다 상태를 새롭게 갱신하여 항상 최신값으로 유지하며, 데이터와 UI에서 입력한 값이 항상 동기화되고 있음을 알 수 있습니다.

비제어 컴포넌트(Uncontrolled Component)는 form의 button을 클릭하기 전까진 값을 알 수 없습니다. 
즉 button을 클릭해야 비로소 값을 가져올 수 있습니다.

따라서 실시간으로 값이 필요할 때는 제어 컴포넌트(Controlled Component)를 사용하고, 
불필요한 재렌더링을 줄이고 제출 시에만 값이 필요할 때는 비제어 컴포넌트(UnControlled Component)를 사용하는 것이 좋습니다.

**Comment by javascriptbiggosoo** [comment]
제어 컴포넌트(Controlled Components)와 비제어 컴포넌트(Uncontrolled Components)는 React에서 폼 입력을 관리하는 두 가지 접근 방식입니다.
제어 컴포넌트는 React 상태(State)에 의해 폼 입력 요소의 값이 제어됩니다. 컴포넌트의 상태가 입력 값의 "진리의 단일 출처" 역할을 하며, 이벤트 핸들러를 통해 상태가 업데이트됩니다.
비제어 컴포넌트는 입력 값이 React 상태(State)가 아닌 HTML DOM에 의해 관리됩니다. 컴포넌트는 ref를 통해 입력 값에 접근할 수 있지만, 값의 변경을 추적하거나 제어하지 않습니다.
제어 컴포넌트는 더 많은 제어권과 유효성 검사, 형식 지정 등의 기능을 제공하며, 비제어 컴포넌트는 간단한 폼이나 외부 라이브러리 통합에 적합합니다.
